app/services/framework_nomis_mappings/nomis_sync_status.rb

```diff
@@ -0,0 +1,32 @@
+module FrameworkNomisMappings
+  class NomisSyncStatus
+    SUCCESS = 'success'.freeze
+    FAILED = 'failed'.freeze
+
+    attr_reader :resource_type, :status, :synced_at, :message
+
+    def initialize(resource_type:)
+      @resource_type = resource_type
+    end
+
+    def set_success
+      @status = SUCCESS
+      @synced_at = Time.zone.now
+    end
+
+    def set_failure(message: nil)
+      @status = FAILED
+      @synced_at = Time.zone.now
+      @message = message
+    end
+
+    def as_json(_options = {})
```

Comment:
You could remove this method and include the `ActiveModel::Model` module instead which would give the same thing for free. Only difference is that the attributes would have stringified names rather than symbols, but I think this is expected behaviour - or at least is how conventional models work.

Reply:
talked in person: this approach will hide keys with no values, so agreed to keep current implementation